app/src/main/java/com/example/parkingsystem/mvp/model/ParkingSpaceReservationModel.java

```diff
@@ -1,61 +1,73 @@
+        verification = new ReservationVerification(database);
     }
 
     @Override
-    public String getDateStart() {
-        return this.reservation.getDateStart();
+    public boolean getDateStartButtonPressed(){
+        return isDateStartButtonPressed;
     }
 
     @Override
-    public void setDateStart(String date) {
-        this.reservation.setDateStart(date);
+    public void setDateStartButtonPressed(boolean isDateStartButtonPressed){
+        this.isDateStartButtonPressed=isDateStartButtonPressed;
     }
 
     @Override
-    public String getTimeStart() {
-        return reservation.getTimeStart();
+    public boolean isPressed(Button btn){
+        return btn.isPressed();
     }
 
     @Override
-    public void setTimeStart(String time) {
-        this.reservation.setTimeStart(time);
+    public Reservation getReservation() {
+        return reservation;
     }
 
     @Override
-    public String getDateEnd() {
-        return reservation.getDateEnd();
-    }
-
-    @Override
-    public void setDateEnd(String date) {
-        this.reservation.setDateEnd(date);
-    }
-
-    @Override
-    public String getTimeEnd() {
-        return reservation.getTimeEnd();
+    public void completeReservationInfo(int parkingSpace, int securityCode) {
+        reservation.setParkingSpace(parkingSpace);
+        reservation.setSecurityCode(securityCode);
     }
 
     @Override
-    public void setTimeEnd(String time) {
-        this.reservation.setTimeEnd(time);
+    public void makeReservation() {
+        database.addReservation(reservation);
     }
 
     @Override
-    public void makeReservation(int parkingSpace, int securityCode) {
-        reservation.setParkingSpace(parkingSpace);
-        reservation.setSecurityCode(securityCode);
-        database.addReservation(reservation);
+    public ReservationVerificationResult checkFields(Reservation reservation) {
+        if (getReservation().getDateStart() == null) {
+            return ReservationVerificationResult.MISSING_DATE_START;
+        } else if (getReservation().getTimeStart() == null) {
+            return ReservationVerificationResult.MISSING_TIME_START;
+        } else if (getReservation().getDateEnd() == null) {
+            return ReservationVerificationResult.MISSING_DATE_END;
+        } else if (getReservation().getTimeEnd() == null) {
+            return ReservationVerificationResult.MISSING_TIME_END;
+        } else if (getReservation().getParkingSpace() == Constants.NUMBER_MINUS_ONE) {
+            return ReservationVerificationResult.MISSING_PARKING_SPACE;
+        } else if (getReservation().getSecurityCode() == Constants.NUMBER_MINUS_ONE) {
+            return ReservationVerificationResult.MISSING_SECURITY_CODE;
+        } else if (!verification.canBeReserved(reservation)) {
+            return ReservationVerificationResult.RESERVATION_OVERLAPPING;
+        } else {
+            return ReservationVerificationResult.SUCCESS;
+        }
```

Comment:
with guard clauses it will look better.
https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
Done!